### PR TITLE
Semant updates

### PR DESCRIPTION
**Pull request by 20joshuaz**
Not done yet

### CODE REVIEW

File: src/ast.ml [hunk 1]
```diff
@@ -32,14 +32,14 @@ type uop =
...
     NumberLit of float 
   | BoolLit of bool 
-  | StringLit of string 
   | CharLit of char 
+  | StringLit of string 
   | ListLit of expr list 
   | Id of string 
   | Binop of expr * bop * expr
   | Unop of uop * expr
+  | Elem of expr * expr
```
Comment (Akash-Nayar) [inline src/ast.ml:41]: might be missing something but what's the significance of `expr * expr` over `string * expr`?

File: src/parser.mly [hunk 2]
```diff
@@ -88,8 +88,8 @@ expr:
   | expr OR expr                   { Binop ($1, Or, $3) }
   | NOT expr                       { Unop (Not, $2) }
   | LPAREN expr RPAREN             { $2 }
+  | expr LSQUARE expr RSQUARE      { Elem ($1, $3) }
```
Comment (Akash-Nayar) [inline src/parser.mly:91]: what's an example of this usage? `expr[expr]` is like `5[5]`?

File: src/sast.ml [hunk 3]
```diff
@@ -4,11 +4,13 @@ type sexpr = dtype * sx
 and sx = 
     SNumberLit of float
   | SBoolLit of bool
-  | SStringLit of string
   | SCharLit of char
+  | SStringLit of string
+  | SListList of sexpr list
```
Comment (Akash-Nayar) [inline src/sast.ml:9]: should this be `SListLit` instead of `SListList`?

File: src/test_src/numbers.bruh [hunk 4]
```diff
@@ -1,9 +1,9 @@
 number a is 5.
 number b is 3.5.
 
-say a + b. # should say 8.5
+say a + b. /* should say 8.5 */
```
Comment (Akash-Nayar) [inline src/test_src/numbers.bruh:4]: i guess it should be `# should say 8.5 #` right?

File: src/test_src/numbers.bruh [hunk 5]
```diff
@@ -1,9 +1,9 @@
...
 
 number foo is 3.
 number baz is 6.
 number bar is bas.
-say foo * bar + baz. # should print 24, might need parenthesis
+say foo * bar + baz. /* should print 24, might need parenthesis */
```
Comment (Akash-Nayar) [inline src/test_src/numbers.bruh:9]: ditto

File: src/test_src/semantinput.bruh [hunk 6]
```diff
@@ -1,9 +1,13 @@
 number a is 1.
 number b is 2.
 
+
 define foo(number a -> number): 
   loop a < 5: 
     a is a + 1.
+
+  /* multiline
```
Comment (Akash-Nayar) [inline src/test_src/semantinput.bruh:9]: i realize these are changes from my old pr but ditto

File: src/test_src/sigwhitespaceinput.bruh [hunk 7]
```diff
@@ -25,19 +25,14 @@ else:
...
     loop k in 1 to 10:
-      sum is sum + i + j + k.  # this comment only works bc there's 1 space after period, comments in general are broken
+      sum is sum + i + j + k.
 number x is 5.
 
-# TODO comment after : don't work
+/* EVERYTHING ABOVE SHOULD BE CORRECT */
 
-# EVERYTHING ABOVE SHOULD BE CORRECT
+/* uncomment next line to generate parser error */
```
Comment (Akash-Nayar) [inline src/test_src/sigwhitespaceinput.bruh:33]: same comment stuff

**Review by Akash-Nayar — CHANGES_REQUESTED** [review]

File: src/ast.ml (hunk 1 above)
Comment (20joshuaz) [reply src/ast.ml:41]: We need to figure out list literals, but with the current syntax, `expr * expr` would support something like `[1, 2, 3][0]`

File: src/parser.mly (hunk 2 above)
Comment (20joshuaz) [reply src/parser.mly:91]: see above

File: src/sast.ml (hunk 3 above)
Comment (20joshuaz) [reply src/sast.ml:9]: yessir, will fix

File: src/test_src/numbers.bruh (hunk 4 above)
Comment (20joshuaz) [reply src/test_src/numbers.bruh:4]: yeah, hmm I didn't change this file, must've been some weird Git thing during rebasing

**Pull request closed without merging**